Code/Python/queue.py

```diff
@@ -0,0 +1,134 @@
+"""
+Problem Statement:
+Add implementation of queue in Python.
+
+Queue: A queue is a linear data-structure where elements avn be inserted and deleted from both the ends.
+
+A queue has the following properties.
+
+• It is opened at both the ends.
+This is the reason why insertion and deletion in a queue is done at different ends.
+
+• The end at which elements are addded in a queue is called the rear/back/tail.
+
+• The end from which elements are removed from the queue is called head/front.
+
+• Queue follows FIFO/LILO methodology to insert/remove elements.
+That is, elements inserted first, will be removed first, and the elements added at last will be removed at last.
+
+• The process of adding elements to the queue is known as 'enqueue'.
+And the process of removing elements from the queue is known as 'dequeue'.
+
+• Unlike lists or arrays, queues typically don't allow for random access to the objects they contain.
+
+Main Operations in Queue:
+• Enqueue: Used to add items to the queue. If the queue is full, it is a condition of the Queue Overflow. Time complexity: O(1).
+
+• Dequeue: Used to remove an element from the queue.If the queue is empty, it is a condition of the Queue Underflow. Time complexity: O(1).
+
+• Front/Peek: Displays an element from the front end. Time complexity: O(1).
+
+• Rear: Displays an element from the rear end. Time complexity: O(1).
+
+• isEmpty: Returns True if the Queue is empty, otherwise, False.
+"""
+
+
+class queue:
+    def __init__(self):
+        self.__queue = []
+        self.__size = 0
+
+    def enqueue(self, data):
+        """
+        adds items to the queue
+        Time-Complexity=O(1)
+        """
+        self.__queue.append(data)
+        self.__size += 1
+
+    def dequeue(self):
+        """
+        remove items from the queue
+        Time-Complexity=O(1)
+        """
+        if not self.isEmpty():
+            element = (self.__queue[0])
+            self.__queue.remove(element)
+            self.__size -= 1
+            print(element, "has been removed")
+            return
+
+    def front(self):
+        """
+        displays an item from the front end
+        Time-Complexity=O(1)
+        """
+        print("Front element is:", self.__queue[0])
+        return
+
+    def rear(self):
+        """
+        displays an item from the rear end
+        Time-Complexity=O(1)
+        """
+        print("Rear element is :", self.__queue[-1])
+        return
+
+    def isEmpty(self):
+        """
+        returns true if the queue is empty.
+        Time-Complexity=O(1)
+        """
+        return self.__size == 0
+
+    def display_queue(self):
+        """
+        displays the queue
+        Time-Complexity=O(n) ;n=size of the queue
```

Comment:
Mention the time complexity at the end, not in between the code

Reply:
Since this is the implementation of queue from scratch, it contains different functions and every function has a different time-complexity.
Do I need to mention the time-complexity of every function implemented in the queue at the end, or for a specific function only..?

Reply:
Mention the time-complexity of every function at the end

Reply:
Okay